app/charts/shared/chart-state.ts

```diff
@@ -810,22 +811,47 @@ export const useChartData = (
     sortData,
     chartConfig,
     timeRangeDimensionId,
+    axisDimensionId,
+    limits,
     getAxisValueAsDate,
     getSegmentAbbreviationOrLabel,
     getTimeRangeDate,
   }: {
     sortData?: (data: Observation[]) => Observation[];
     chartConfig: ChartConfig;
     timeRangeDimensionId: string | undefined;
+    axisDimensionId?: string;
+    limits?: Limit[];
     getAxisValueAsDate?: (d: Observation) => Date;
     getSegmentAbbreviationOrLabel?: (d: Observation) => string;
     getTimeRangeDate?: (d: Observation) => Date;
   }
 ): ChartStateData => {
   const { interactiveFiltersConfig } = chartConfig;
   const allData = useMemo(() => {
-    return sortData ? sortData(observations) : observations;
-  }, [observations, sortData]);
+    let allData = observations;
+
+    if (axisDimensionId && limits) {
+      const dimensionValuesObservations = limits
+        .flatMap((limit) => limit.related)
+        .filter((limit) => limit.dimensionId === axisDimensionId)
```

Comment:
Nit: wonder if we should keep a "limit" object indexed by dimension id.

Reply:
Hmmm, in this place I am not 100% sure – generally we need to have easy access to all limits, not scoped to `dimensionId`, e.g. in color legend, to list them all. As it's usually a small number (in the cubes I worked with it was max 4 limits overall), maybe it would be fine to leave as is? 🤔 Otherwise it would either introduce a bit more complexity in other places to flatten the indexed object to an array, or require keeping both indexed object and an array 🥹

Reply:
Fine by me :)